src/core/mac/mac.cpp

```diff
@@ -980,111 +980,61 @@ extern "C" void otPlatRadioTxDone(otInstance *aInstance, otRadioFrame *aFrame, o
     else
 #endif // OPENTHREAD_ENABLE_RAW_LINK_API
     {
-        instance->GetThreadNetif().GetMac().TransmitDoneTask(aFrame, aAckFrame, aError);
+        instance->GetThreadNetif().GetMac().HandleTransmitDone(aFrame, aAckFrame, aError);
     }
 
 exit:
     otLogFuncExit();
 }
 
-void Mac::TransmitDoneTask(otRadioFrame *aFrame, otRadioFrame *aAckFrame, otError aError)
+void Mac::HandleTransmitDone(otRadioFrame *aFrame, otRadioFrame *aAckFrame, otError aError)
 {
-    Frame *txFrame = static_cast<Frame *>(aFrame);
-    Address addr;
+    Frame &sendFrame = *static_cast<Frame *>(aFrame);
+    Address dstAddr;
     bool framePending = false;
+    Neighbor *neighbor = NULL;
 
+    // Stop the ack timer.
     mMacTimer.Stop();
 
-    txFrame->GetDstAddr(addr);
-
-    if (aError == OT_ERROR_NONE && txFrame->GetAckRequest() && aAckFrame != NULL)
-    {
-        Frame *ackFrame = static_cast<Frame *>(aAckFrame);
-        Neighbor *neighbor;
-
-        framePending = ackFrame->GetFramePending();
-        neighbor = GetNetif().GetMle().GetNeighbor(addr);
-
-        if (neighbor != NULL)
-        {
-            neighbor->GetLinkInfo().AddRss(GetNoiseFloor(), ackFrame->GetRssi());
-        }
-    }
-
-    if (aError == OT_ERROR_ABORT)
+    switch (aError)
     {
+    case OT_ERROR_ABORT:
         mCounters.mTxErrAbort++;
-    }
 
-    if (aError == OT_ERROR_CHANNEL_ACCESS_FAILURE)
-    {
-        mCounters.mTxErrCca++;
-    }
+    // fall through
 
-    if (!RadioSupportsCsmaBackoff() &&
-        aError == OT_ERROR_CHANNEL_ACCESS_FAILURE &&
-        mCsmaAttempts < kMaxCSMABackoffs)
-    {
-        mCsmaAttempts++;
-        StartCsmaBackoff();
+    case OT_ERROR_NONE:
+    case OT_ERROR_NO_ACK:
+    case OT_ERROR_CHANNEL_ACCESS_FAILURE:
+        break;
 
+    default:
+        assert(false);
         ExitNow();
     }
 
-    mCsmaAttempts = 0;
+    // Determine whether a CSMA retry is required.
 
-    switch (mOperation)
+    if (aError == OT_ERROR_CHANNEL_ACCESS_FAILURE)
     {
-    case kOperationTransmitData:
-
-        if (framePending && txFrame->IsDataRequestCommand())
-        {
-            mReceiveTimer.Start(kDataPollTimeout);
-            StartOperation(kOperationWaitingForData);
-        }
-
-    // fall through
-
-    case kOperationActiveScan:
-    case kOperationTransmitBeacon:
-        SentFrame(aError);
-        break;
+        mCounters.mTxErrCca++;
 
-    default:
-#if OPENTHREAD_ENABLE_RAW_LINK_API
```

Comment:
This logic is removed (since there was no way to get to this path. If `OPENTHREAD_ENABLE_RAW_LINK_API` is enabled the `HandleTrasnmitDone()`  won't be invoked.